From 9ba775ea21b20ce57106fc508f33137e54fbb90d Mon Sep 17 00:00:00 2001
From: Kyle Brenneman <kbrenneman@nvidia.com>
Date: Fri, 22 Nov 2019 16:36:04 -0700
Subject: [PATCH] configure: Define EGL_NO_X11 unconditionally.

Change the configure script to define the EGL_NO_X11 macro unconditionally, so
that the EGL headers don't try to include the Xlib headers.

Regardless of whether --disable-x11 is used, libglvnd doesn't need X11-specific
typedefs for EGLNativeDisplayType et. al.
---
 configure.ac | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/configure.ac b/configure.ac
index 1b4074d..199d92e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -366,6 +366,10 @@ AS_IF([test "x$GLDISPATCH_PAGE_SIZE" != "x"],
       [AC_DEFINE_UNQUOTED([GLDISPATCH_PAGE_SIZE], [$GLDISPATCH_PAGE_SIZE],
       [Page size to align static dispatch stubs.])])
 
+# Set EGL_NO_X11 unconditionally. Libglvnd doesn't make any assumptions about
+# native display or drawable types, so we don't need X11-specific typedefs for
+# them.
+DEFINES="$DEFINES -DEGL_NO_X11"
 
 dnl default CFLAGS
 CFLAGS="$CFLAGS -Wall -Werror -include config.h -fvisibility=hidden $DEFINES"
-- 
2.24.1

